Fix Desync Error, Fixes #1

Nexela opened the pull request:
Set up global table for per user usage
Fix logic when un-slowing in case something else changed player modifier
Add text overlay when slowed
Add remote interfaces for setting/reading the modifier value
Increment version

Nexela committed:
Fix Desync Error, Fixes #1
Set up global table for per user usage
Fix logic when un-slowing in case something else changed player modifier
Add text overlay when slowed
Add remote interfaces for setting/reading the modifier value
Increment version

lf- commented:
Needs some comment spelling+grammar fixes.

lf- commented on control.lua:
```diff
@@ -1,21 +1,102 @@
+
+--create the global if not exists, Doing this here will have no noticeable impact on performance.
+local function get_or_create_global(player_index)
+  if not global.player_data then global.player_data = {} end
```
global.player_data = global.player_data or {}

Nexela replied:
Redone, however either one should work with the same efficiency

lf- commented on control.lua:
```diff
@@ -1,21 +1,102 @@
+  player = get_valid_player(player)
+  if player then
+    local pdata = get_or_create_global(player.index)
+    player.print("slow-key: Current modifier is "..pdata.modifier)
```
Spacing around operators: `..`

Nexela replied:
Made all spacing consistent.

lf- commented on control.lua:
```diff
@@ -1,21 +1,102 @@
+      else -- We are at normal speed, lets slow down
+        pdata.last_modifier = cur_modifier -- Save a copy of the players current modifier in case it gets changed.
+        pdata.saved_modifier = pdata.modifier -- Save a copy of the slow modifier in case we change it.
+        pdata.difference = cur_modifier - pdata.modifier -- get the difference in case players modifier changes some other way.
```
Inconsistent spacing around comment start character & first character in comment capitalization

Nexela replied:
I blame it on split personalities :)

lf- commented on control.lua:
```diff
@@ -1,21 +1,102 @@
 -- Bounds for expected behaviour: 0.0 > x > -1.0
--- Above 0: becomes FastKey
+-- Above 0: becomes FastKey (0 is default modifier when creating player unless changed by mods)
```
Addition should be on a separate line? Is this even true? `MODIFIER = -0.5` further down in the file

Nexela replied:
Moved, and clarified

lf- commented on control.lua:
```diff
@@ -1,21 +1,102 @@
+      else -- We are at normal speed, lets slow down
+        pdata.last_modifier = cur_modifier -- Save a copy of the players current modifier in case it gets changed.
+        pdata.saved_modifier = pdata.modifier -- Save a copy of the slow modifier in case we change it.
+        pdata.difference = cur_modifier - pdata.modifier -- get the difference in case players modifier changes some other way.
```
Why is it necessary to store difference? It's 100% unused as far as actual functionality, the variable should probably be removed.

Nexela replied:
Thinking some more on this is it might not be easy to reset

Example 1
Toggle slow-key on
MobB does modifier=modifier+arbitrary number

Example 2
Toggle slow-key on
ModC be does modifier = number

No matter which way we go with it, After toggling off we could have incorrect values.

I am going to account for example 2 and just disable slow-key without changing if the modifier is not the same as the saved one. Let me know if you have any better ideas.

lf- reviewed (changes requested):
Pretty good, just needs a couple code style/nitpick changes

Nexela committed:
Add addition sanity checking to get_valid_player
Changed mod name strings to be consistent with the mods name/title
Syntax changes
Change from player.character to just player
Remove connected and character checks to controllers check
Trim the fat from the meat and potatoes
Add Interface print_global, and change_modifier_for_all
Rename Interfaces
Consistent spacing/punction throughout.

Nexela committed:
Typo fixes

Nexela reviewed (commented):
All changes made, played with it in game for a couple of days with no issues.

lf- commented:
Awesome, merging
